spec/components/trainees/confirmation/programme_details/view_spec.rb

```diff
@@ -0,0 +1,52 @@
+require "rails_helper"
+
+module Trainees
+  module Confirmation
+    module ProgrammeDetails
+      describe View do
+        alias_method :component, :page
+
+        context "when data has not been provided" do
+          let(:trainee) { build(:trainee, id: 1, subject: nil, age_range: nil, programme_start_date: nil) }
+
+          before do
+            render_inline(View.new(trainee: trainee))
+          end
+
+          it "renders blank rows for subject, age range, and programme start date" do
+            expect(component).to have_selector(".govuk-summary-list__row", count: 3)
+          end
+
+          it "tells the user that no data has been entered" do
+            component.find_all(".govuk-summary-list__row").each do |row|
+              expect(row.find(".govuk-summary-list__value")).to have_text(t("components.confirmation.not_provided"))
+            end
+          end
```

Comment:
```suggestion

          it "tells the user that no data has been entered for subject, age range, and programme start date" do
            found = component.find_all(".govuk-summary-list__row")
            
            expect(found).to have size 3
            
            found.each do |row|
              expect(row.find(".govuk-summary-list__value")).to have_text(t("components.confirmation.not_provided"))
            end
          end
```
``` ruby
it "renders blank rows for subject, age range, and programme start date" do
# seems a bit pointless
```

Reply:
Done, although this little bit isn't as functional as you implied! `expect(found).to have size 3`